### PR TITLE
feat: add enable cache buster

### PR DESCRIPTION
**Pull request by nattaaek**

### CODE REVIEW

File: dashboard-plugin/client-version.js [hunk 1]
```diff
@@ -5,16 +5,18 @@ module.exports = ({
   fallbackEntryURL,
   fallbackRemoteVar,
   dashboardTimeout,
+  enableCacheBuster,
 }) => {
-  fallbackRemoteVar = fallbackRemoteVar || remoteName;
-  fallbackEntryURL = fallbackEntryURL || '';
+  fallbackRemoteVar = fallbackRemoteVar || remoteName
+  fallbackEntryURL = fallbackEntryURL || ''
+  enableCacheBuster = enableCacheBuster || false
```
Comment (edusperoni) [inline dashboard-plugin/client-version.js:12]: re-add semicolons. switch to fallbackEnableCacheBuser so it's clear the cachebuster is for fallback only

File: dashboard-plugin/client-version.js [hunk 2]
```diff
@@ -5,16 +5,18 @@ module.exports = ({
...
   if (!dashboardTimeout) {
-    dashboardTimeout = -1;
+      dashboardTimeout = -1
```
Comment (edusperoni) [inline dashboard-plugin/client-version.js:14]: remove extra white space indentation, add semicolon

File: dashboard-plugin/client-version.js [hunk 3]
```diff
@@ -5,16 +5,18 @@ module.exports = ({
...
   }
   if (typeof dashboardTimeout !== 'number' || isNaN(dashboardTimeout)) {
-    throw new Error('Invalid dashboardTimeout');
+      throw new Error('Invalid dashboardTimeout')
```
Comment (edusperoni) [inline dashboard-plugin/client-version.js:17]: remove extra white space indentation, add semicolon

File: dashboard-plugin/client-version.js [hunk 4]
```diff
@@ -5,16 +5,18 @@ module.exports = ({
...
   }
-  //language=JS
+  // language=JS
```
Comment (edusperoni) [inline dashboard-plugin/client-version.js:19]: remove whitespace difference

File: dashboard-plugin/client-version.js [hunk 5]
```diff
@@ -71,10 +73,16 @@ module.exports = ({
...
+          var fallbackEntryURL = '${fallbackEntryURL}';
+          if (!fallbackEntryURL) return Promise.reject(error);
+          var name = '${fallbackRemoteVar}';
+          var url = new URL(fallbackEntryURL);
+          
+          // Generate a unique timestamp or random value and append it as a query parameter
+          if (${enableCacheBuster}) {
+              var cacheBuster = Date.now(); // You can use a timestamp
+              url.searchParams.append('cacheBuster', cacheBuster);
+          }
```
Comment (edusperoni) [inline dashboard-plugin/client-version.js:85]: remove extra (2) whitespaces to make it consistent with the rest of the code in this section

File: dashboard-plugin/client-version.js [hunk 6]
```diff
@@ -100,5 +108,5 @@ module.exports = ({
           })
           .catch(reject);
       });
-  });`;
-};
+  });`
+}
```
Comment (edusperoni) [inline dashboard-plugin/client-version.js:112]: add semicolons

**Review by edusperoni — CHANGES_REQUESTED** [review]
Hello! Thanks for this. I've added some comments to keep the code consistent

**Review by edusperoni — APPROVED** [review]

**Review by valorkin — APPROVED** [review]

**Pull request merged**